modules/post/windows/manage/execute_dotnet_assembly.rb

```diff
@@ -0,0 +1,296 @@
+      'Author' => 'b4rtik',
+      'Arch' => [ARCH_X64],
+      'Platform' => 'win',
+      'SessionTypes' => ['meterpreter'],
+      'Targets' => [['Windows x64 (<= 10)', { 'Arch' => ARCH_X64 }]],
+      'References' => [['URL', 'https://b4rtik.blogspot.com/2018/12/execute-assembly-via-meterpreter-session.html']],
+      'DefaultTarget' => 0
+    ))
+    register_options(
+      [
+        OptPath.new('DOTNET_EXE', [true, 'Assembly file name']),
+        OptString.new('ARGUMENTS', [false, 'Command line arguments']),
+        OptString.new('PROCESS', [false, 'Process to spawn','notepad.exe']),
+        OptString.new('USETHREADTOKEN', [false, 'Spawn process with thread impersonation',true]),
+        OptInt.new('PID', [false, 'Pid  to inject', 0]),
+        OptInt.new('PPID', [false, 'Process Identifier for PPID spoofing when creating a new process. (0 = no PPID spoofing)', 0]),
+        OptBool.new('AMSIBYPASS', [true, 'Enable Amsi bypass', true]),
+        OptBool.new('ETWBYPASS', [true, 'Enable Etw bypass', true]),
+        OptInt.new('WAIT', [false, 'Time in seconds to wait', 10])
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('KILL',   [ true, 'Kill the injected process at the end of the task', false ])
+      ]
+    )
+  end
+  def check_dotnet_version
+    vprint_status("DOTNET VERSIONS:  #{get_dotnet_versions}")
+  end
+
+  def find_required_clr(exe_path)
+    filecontent = File.read(exe_path).bytes
+    sign = "v4.0.30319".bytes
+    filecontent.each_with_index do |item, index|
+      sign.each_with_index do |subitem, indexsub|
+        if subitem.to_s(16) != filecontent[index + indexsub].to_s(16)
+          break
+        else
+          if indexsub == 9
+            vprint_status("CLR versione required v4.0.30319")
+            return "v4.0.30319"
+          end
+        end
+      end
+    end
+    vprint_status("CLR versione required v2.0.50727")
+    return "v2.0.50727"
+  end
+
+  def check_requirements(clr_req, installed_dotnet_versions)
+    installed_dotnet_versions.each do | fi |
+      if clr_req == "v4.0.30319"
+        if fi[0] == "4"
+          vprint_status("Requirements ok")
+          return true
+        end
+      else
+        if fi[0] == "3"
+          vprint_status("Requirements ok")
+          return true
+        end
+      end
+    end
+    vprint_status("Requirements ko")
+    return false
+  end
+
+  def run
+    installed_dotnet_versions = get_dotnet_versions
+    vprint_status("Dot Net Versions installed on target: #{installed_dotnet_versions}")
+    if installed_dotnet_versions == []
+      fail_with(Failure::BadConfig, "Target has no .NET framework installed")
+    end
+    exe_path = datastore['DOTNET_EXE']
+    if check_requirements(find_required_clr(exe_path), installed_dotnet_versions) == false
+      fail_with(Failure::BadConfig, "CLR required for assembly not installed")
+    end
+    if File.file?(exe_path)
+      assembly_size = File.size(exe_path)
```

Comment:
Hey; I've just been running rubocop over some files. This variable seems to be unused, was that intentional?

```
modules/post/windows/manage/execute_dotnet_assembly.rb:108:7: W: Lint/UselessAssignment: Useless assignment to variable - assembly_size.
      assembly_size = File.size(exe_path)
      ^^^^^^^^^^^^^
modules/post/windows/manage/execute_dotnet_assembly.rb:110:9: W: Lint/UselessAssignment: Useless assignment to variable - params_size.
        params_size = 0
        ^^^^^^^^^^^
modules/post/windows/manage/execute_dotnet_assembly.rb:112:9: W: Lint/UselessAssignment: Useless assignment to variable - params_size.
        params_size = datastore['ARGUMENTS'].length
        ^^^^^^^^^^^
```

I saw this when running the following command:

```
rubocop modules/post/windows/manage/execute_dotnet_assembly.rb
```

Reply:
No. Probably a copy & paste. This should be:

```
    if File.file?(exe_path)
      execute_assembly(exe_path)
    else
      print_bad("Assembly not found #{exe_path}")
    end
```

How can I proceed to correct it?

Reply:
If you want to, you can check out master and put up another PR with a fix.  Feel free to assign it to me if you can, as I can test/verify pretty quick, now.  It might not be bad to go ahead and run `rubocop -a` against it as well.

Reply:
https://github.com/rapid7/metasploit-framework/pull/13279